src/login/logind.c

```diff
@@ -852,32 +805,35 @@ static int manager_connect_console(Manager *m) {
         return 0;
 }
 
-static int manager_connect_udev(Manager *m) {
-        int r;
+static int manager_dispatch_device_udev(sd_device_monitor *monitor, sd_device *device, void *userdata) {
+        Manager *m = ASSERT_PTR(userdata);
 
-        assert(m);
-        assert(!m->device_seat_monitor);
-        assert(!m->device_monitor);
-        assert(!m->device_vcsa_monitor);
-        assert(!m->device_button_monitor);
+        assert(device);
 
-        r = sd_device_monitor_new(&m->device_seat_monitor);
-        if (r < 0)
-                return r;
+        if (device_in_subsystems(device, STRV_MAKE("input", "graphics", "drm")) ||
+            sd_device_has_current_tag(device, "master-of-seat") > 0)
+                (void) manager_process_seat_device(m, device);
 
-        r = sd_device_monitor_filter_add_match_tag(m->device_seat_monitor, "master-of-seat");
-        if (r < 0)
-                return r;
+        if (!manager_all_buttons_ignored(m) &&
+            device_in_subsystem(device, "input") &&
+            sd_device_has_current_tag(device, "power-switch") > 0)
+                (void) manager_process_button_device(m, device);
 
-        r = sd_device_monitor_attach_event(m->device_seat_monitor, m->event);
-        if (r < 0)
-                return r;
+        /* Whenever a VCSA device is removed try to reallocate our VTs, to make sure our auto VTs never go away. */
+        const char *name;
+        if (device_in_subsystem(device, "vc") &&
+            sd_device_get_sysname(device, &name) >= 0 && startswith(name, "vcsa") &&
```

Comment:
one of those days we should probably start logging about sd_device_get_sysname() failing here...